hrjobcontract/js/src/job-contract/controllers/form/form-general.js

```diff
@@ -3,99 +3,129 @@ define([
-                }
-            });
-
-        }]);
+  'use strict';
+
+  controllers.controller('FormGeneralCtrl',['$scope','$log', 'HR_settings',
+    function ($scope, $log, HR_settings) {
+      $log.debug('Controller: FormGeneralCtrl');
+
+      var entityDetails = $scope.entity.details;
+
+      $scope.format = HR_settings.DATE_FORMAT;
+      $scope.datepickerOptions = initDatepickerOptions();
+
+      $scope.dpOpen = function($event, opened){
+        $event.preventDefault();
+        $event.stopPropagation();
+
+        $scope[opened] = true;
+      };
+
+      $scope.$watch('entity.details.period_start_date', function(){
+        $scope.datepickerOptions.end.minDate = getLimitDate(entityDetails.period_start_date, 'min');
+        $scope.duration = duration(entityDetails.period_start_date, entityDetails.period_end_date);
+      });
+
+      $scope.$watch('entity.details.period_end_date', function(){
+        if (entityDetails.period_end_date) {
+          $scope.datepickerOptions.start.maxDate = getLimitDate(entityDetails.period_end_date, 'max');
+        } else {
+          $scope.datepickerOptions.start.maxDate = null;
+          entityDetails.end_reason = null;
+        }
+
+        $scope.duration = duration(entityDetails.period_start_date, entityDetails.period_end_date);
+      });
+
+      $scope.$watch('entity.details.position', function(newVal, oldVal){
+        if (newVal != oldVal && entityDetails.title == oldVal) {
+          $scope.contractForm.detailsTitle.$setViewValue(newVal);
+          $scope.contractForm.detailsTitle.$render();
+        }
+      });
+
+      $scope.$watch('entity.details.notice_amount', function(newVal, oldVal){
+        if (+newVal && !entityDetails.notice_unit) {
+          $scope.contractForm.detailsNoticeUnit.$setValidity('required', false);
+          $scope.contractForm.detailsNoticeUnit.$dirty = true;
+        }
+
+        if (newVal != oldVal && entityDetails.notice_amount_employee == oldVal) {
+          entityDetails.notice_amount_employee = newVal;
+        }
+      });
+
+      $scope.$watch('entity.details.notice_amount_employee', function(newVal){
+        if (+newVal && !entityDetails.notice_unit_employee) {
+          $scope.contractForm.detailsNoticeUnitEmployee.$setValidity('required', false);
+          $scope.contractForm.detailsNoticeUnitEmployee.$dirty = true;
+        }
+      });
+
+      $scope.$watch('entity.details.notice_unit', function(newVal, oldVal){
+        if (newVal != oldVal && entityDetails.notice_unit_employee == oldVal) {
+          entityDetails.notice_unit_employee = newVal;
+        }
+      });
+
+      /**
+       * Calculates the duration of the period between the given start and end dates
+       *
+       * @param  {Date} dateStart
+       * @param  {Date} dateEnd
+       * @return {string}
+       */
+      function duration(dateStart, dateEnd){
+        if (!dateStart || !dateEnd) {
+          return 'Unspecified';
```

Comment:
I've moved this string in the template (the view), where it should've belonged from the start